app/ui.js

```diff
@@ -39,22 +39,20 @@ const UI = {
     reconnect_callback: null,
     reconnect_password: null,
 
-    prime(callback) {
+    prime() {
         if (document.readyState === "interactive" || document.readyState === "complete") {
-            UI.load(callback);
-        } else {
-            document.addEventListener('DOMContentLoaded', UI.load.bind(UI, callback));
+            return WebUtil.initSettings().then(UI.start);
```

Comment:
Some duplication of `WebUtil.initSettings()` that might be possible to chain away. But otherwise it looks very nice. :)